File: travel_expense_backend/src/modules/travel/entities/travel-request.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, ManyToOne, JoinColumn, OneToMany } from 'typeorm';
import { User } from '../../users/entities/user.entity';
import { ApprovalStage } from '../../approvals/entities/approval-stage.entity';

export enum TravelStatus {
  DRAFT = 'draft',
  SUBMITTED = 'submitted',
  PENDING_L1 = 'pending_l1',
  PENDING_L2 = 'pending_l2',
  PENDING_L3 = 'pending_l3',
  APPROVED = 'approved',
  BOOKED = 'booked',
  TRAVEL_COMPLETED = 'travel_completed',
  CLAIM_DRAFT = 'claim_draft',
  CLAIM_SUBMITTED = 'claim_submitted',
  CLAIM_FINANCE_APPROVED = 'claim_finance_approved',
  REIMBURSED = 'reimbursed',
  REJECTED = 'rejected',
  CANCELLED = 'cancelled',
}

@Entity('travel_requests')
export class TravelRequest {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  title: string;

  @Column({ type: 'text', nullable: true })
  description: string;

  @Column()
  purpose: string;

  @Column()
  destination: string;

  @Column()
  startDate: Date;

  @Column()
  endDate: Date;

  @Column({ type: 'decimal', precision: 12, scale: 2, default: 0.0 })
  estimatedCost: number;

  @Column({
    type: process.env.OFFLINE_MODE === 'true' ? 'simple-enum' : 'enum',
    enum: TravelStatus,
    default: TravelStatus.DRAFT,
  })
  status: TravelStatus;

  @Column()
  userId: string;

  @ManyToOne(() => User, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'userId' })
  user: User;

  @OneToMany(() => ApprovalStage, (stage) => stage.travelRequest)
  approvalStages: ApprovalStage[];

  @CreateDateColumn()
  createdAt: Date;
}

File: travel_expense_app/lib/features/travel_request/domain/entities/travel_request_entity.dart
import 'package:equatable/equatable.dart';

class TravelRequestEntity extends Equatable {
  final String id;
  final String title;
  final String description;
  final String purpose;
  final String destination;
  final DateTime startDate;
  final DateTime endDate;
  final double estimatedCost;
  final String status; // 'pending', 'approved', 'rejected', 'cancelled'
  final DateTime createdAt;

  const TravelRequestEntity({
    required this.id,
    required this.title,
    required this.description,
    required this.purpose,
    required this.destination,
    required this.startDate,
    required this.endDate,
    required this.estimatedCost,
    required this.status,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [
        id,
        title,
        description,
        purpose,
        destination,
        startDate,
        endDate,
        estimatedCost,
        status,
        createdAt,
      ];
}
